docker/recorder/src/main/java/org/arquillian/cube/docker/impl/client/reporter/TakeDockerEnvironment.java

```diff
@@ -58,9 +75,66 @@ public void reportDockerEnvironment(@Observes AfterAutoStart event, CubeDockerCo
         docker.getPropertyEntries().add(containersComposition);
 
         propertyReportEvent.fire(new PropertyReportEvent(docker));
+    }
+
+    public void reportContainerStatsBeforeTest(@Observes Before before, DockerClientExecutor executor) {
+        generateStats(executor, false, "Before");
+    }
+
+    public void reportContainerStatsAfterTest(@Observes After after, DockerClientExecutor executor, ReporterConfiguration configuration) {
+        generateStats(executor, false, "After");
+    }
+
+    private void generateStats(DockerClientExecutor executor, Boolean decimal, String when) {
+        if (executor != null) {
+            List<Cube<?>> containers = cubeRegistryInstance.get().getCubes();
 
+            GroupEntry containersStat = new GroupEntry("Container Statistics " + when + " Method");
+
+            for (Cube<?> container : containers) {
+                String name = container.getId();
+                Statistics statistics = executor.statsContainer(name);
+                Map<String, Map> stats = getStats(statistics, decimal);
```

Comment:
What's the reason behind using raw `Map` type here? Can't we do it more elegant?